ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -358,27 +373,328 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
+            auto const& dNdx = ip_data.dNdx;
+            auto const& w = ip_data.integration_weight;
+
+            double C_int_pt = 0.0;
+            double p_int_pt = 0.0;
+
+            NumLib::shapeFunctionInterpolate(local_C, N, C_int_pt);
+            NumLib::shapeFunctionInterpolate(local_p, N, p_int_pt);
+
+            // porosity model
+            auto const porosity =
+                _process_data.porous_media_properties.getPorosity(t, pos)
+                    .getValue(t, pos, 0.0, C_int_pt);
+
+            auto const retardation_factor =
+                _process_data.retardation_factor(t, pos)[0];
+
+            auto const& solute_dispersivity_transverse =
+                medium.template value<double>(
+                    MaterialPropertyLib::transversal_dispersivity);
+            auto const& solute_dispersivity_longitudinal =
+                medium.template value<double>(
+                    MaterialPropertyLib::longitudinal_dispersivity);
+
+            // Use the fluid density model to compute the density
+            vars[static_cast<int>(
+                MaterialLib::Fluid::PropertyVariableType::C)] = C_int_pt;
+            vars[static_cast<int>(
+                MaterialLib::Fluid::PropertyVariableType::p)] = p_int_pt;
+            auto const density = _process_data.fluid_properties->getValue(
+                MaterialLib::Fluid::FluidPropertyType::Density, vars);
+            auto const decay_rate = _process_data.decay_rate(t, pos)[0];
+
+            auto const& molecular_diffusion_coefficient =
+                component.template value<double>(
+                    MaterialPropertyLib::molecular_diffusion);
+
+            auto const& K =
+                _process_data.porous_media_properties.getIntrinsicPermeability(
+                    t, pos).getValue(t, pos, 0.0, 0.0);
+            // Use the viscosity model to compute the viscosity
+            auto const mu = _process_data.fluid_properties->getValue(
+                MaterialLib::Fluid::FluidPropertyType::Viscosity, vars);
+
+            GlobalDimMatrixType const K_over_mu = K / mu;
+            GlobalDimVectorType const velocity =
+                _process_data.has_gravity
+                    ? GlobalDimVectorType(-K_over_mu *
+                                          (dNdx * local_p - density * b))
+                    : GlobalDimVectorType(-K_over_mu * dNdx * local_p);
+
+            const double drho_dp = _process_data.fluid_properties->getdValue(
+                MaterialLib::Fluid::FluidPropertyType::Density,
+                vars,
+                MaterialLib::Fluid::PropertyVariableType::p);
+
+            const double drho_dC = _process_data.fluid_properties->getdValue(
+                MaterialLib::Fluid::FluidPropertyType::Density,
+                vars,
+                MaterialLib::Fluid::PropertyVariableType::C);
+            double const velocity_magnitude = velocity.norm();
+            GlobalDimMatrixType const hydrodynamic_dispersion =
```

Comment:
The definition repeats what for the monolithic scheme:
https://github.com/renchao-lu/ogs6/blob/MultiComponentTransportProcess_SS/ProcessLib/ComponentTransport/ComponentTransportFEM.h#L328

Reply:
For sure somewhat code duplication. So far I have no good idea on how to prevent duplication of this kind.

Reply:
You can introduce a member function including this part together with lines 566-571, 
```

```

Reply:
Done.

Reply:
Remove the commit for processing this issue which is not on propose to this pull request. I prefer to set up another pull request later on to resolve this particular issue.